lambdas/data-migration2/src/granulesAndFiles.ts

```diff
@@ -0,0 +1,208 @@
+import Knex from 'knex';
+
+import DynamoDbSearchQueue from '@cumulus/aws-client/DynamoDbSearchQueue';
+import { ApiFile } from '@cumulus/types/api/files';
+import {
+  CollectionPgModel,
+  GranulePgModel,
+  PostgresFile,
+  PostgresGranuleRecord,
+  tableNames,
+  translateApiGranuleToPostgresGranule,
+} from '@cumulus/db';
+import { envUtils } from '@cumulus/common';
+import Logger from '@cumulus/logger';
+
+import { RecordAlreadyMigrated } from '@cumulus/errors';
+import { MigrationSummary } from './types';
+
+const logger = new Logger({ sender: '@cumulus/data-migration/granules' });
+const Manager = require('@cumulus/api/models/base');
+const schemas = require('@cumulus/api/models/schemas');
+const { getBucket, getKey } = require('@cumulus/api/lib/FileUtils');
+const { deconstructCollectionId } = require('@cumulus/api/lib/utils');
+
+export interface GranulesAndFilesMigrationSummary {
+  granulesSummary: MigrationSummary,
+  filesSummary: MigrationSummary,
+}
+
+/**
+ * Migrate granules record from Dynamo to RDS.
+ *
+ * @param {AWS.DynamoDB.DocumentClient.AttributeMap} record
+ *   Record from DynamoDB
+ * @param {Knex} knex - Knex client for writing to RDS database
+ * @returns {Promise<void>}
+ * @throws {RecordAlreadyMigrated} if record was already migrated
+ */
+export const migrateGranuleRecord = async (
+  record: AWS.DynamoDB.DocumentClient.AttributeMap,
+  knex: Knex
+): Promise<void> => {
+  // Validate record before processing using API model schema
+  Manager.recordIsValid(record, schemas.granule);
+  const { name, version } = deconstructCollectionId(record.collectionId);
+  const collectionPgModel = new CollectionPgModel();
+
+  const collectionCumulusId = await collectionPgModel.getRecordCumulusId(
+    knex,
+    { name, version }
+  );
+
+  const existingRecord = await knex<PostgresGranuleRecord>('granules')
+    .where({ granule_id: record.granuleId, collection_cumulus_id: collectionCumulusId })
+    .first();
+
+  // Throw error if it was already migrated.
+  if (existingRecord) {
+    throw new RecordAlreadyMigrated(`Granule ${record.granuleId} was already migrated, skipping`);
+  }
+
+  const granule = await translateApiGranuleToPostgresGranule(record, knex, collectionCumulusId);
+  await knex(tableNames.granules).insert(granule);
```

Comment:
Should also be using a granule model method, possibly for `upsert`

Reply:
Why possibly? It seems like we should use upsert [`c8ed8f2` (#2051)](https://github.com/nasa/cumulus/pull/2051/commits/c8ed8f2940a7d3697b4eb7f15c857f5892656ae9)

Reply:
It occurs to me that as long as we have logic to skip records if they already exist in the granules/files tables, then there's no point in using `upsert`, because it'll never try to update an existing record. But that's a question outside the scope of this work and PR